crates/network/src/transaction/builder.rs

```diff
@@ -227,8 +230,24 @@ pub trait TransactionBuilder<N: Network>: Default + Sized + Send + Sync + 'stati
     /// a valid transaction.
     fn can_build(&self) -> bool;
 
+    /// Returns the transaction type that this builder will attempt to build.
+    /// This does not imply that the builder is ready to build.
+    fn output_tx_type(&self) -> TxType;
```

Comment:
hmm, can we use the txtype in the trait here?

because txtypes are network specific

could operate on u8 or make txtype part of the network trait

Reply:
good catch, we cannot. 

> could operate on u8 or make txtype part of the network trait

I am partial to the idea of using `TxType` on the network trait, but at that point, i would also want to incorporate into `En/Decodable2718`, and make something like `trait En/Decodable2718<T: Into<u8> + TryFrom<u8, Error = _>>` or similar so that we can enforce that the 2718 implementation uses the network `TxType` via a `type TxEnvelope: Encodable2718<Self::TxType>` bound 

let me think about it a bit. most likely outcome is that this PR will operate on u8, and i will reserve network-assoc txtype for a later pr

Reply:
okay i made a network specific tx type because it felt right >.>